Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryTransforms.java

```diff
@@ -0,0 +1,247 @@
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import gov.hhs.fha.nhinc.audit.transform.AuditTransform;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.patientdiscovery.audit.PatientDiscoveryTransformConstants;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import gov.hhs.fha.nhinc.util.HomeCommunityMap;
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.hl7.v3.II;
+import org.hl7.v3.PRPAIN201305UV02;
+import org.hl7.v3.PRPAIN201306UV02;
+import org.hl7.v3.PRPAIN201306UV02MFMIMT700711UV01Subject1;
+
+/**
+ *
+ * @author achidamb
+ * @param <T>
+ * @param <K>
+ */
+public class PatientDiscoveryTransforms<T extends PRPAIN201305UV02, K extends PRPAIN201306UV02> extends AuditTransform<T, K> {
+
+    private static final Logger LOG = Logger.getLogger(PatientDiscoveryTransforms.class);
+
+    /**
+     *
+     * @param assertion
+     * @param auditMsg
+     * @return
+     */
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForRequest(PRPAIN201305UV02 request, AssertionType assertion,
+        AuditMessageType auditMsg) {
+        auditMsg = getPatientParticipantObjectIdentificationForRequest(request, auditMsg);
+        try {
+            auditMsg = getQueryParamsParticipantObjectIdentificationForRequest(request, auditMsg);
+        } catch (JAXBException ex) {
+            LOG.error("Error while creating ParticipantObjectIdentificationQueryByParameters segment : "
+                + ex.getLocalizedMessage(), ex);
+        }
+        return auditMsg;
+    }
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForResponse(PRPAIN201306UV02 response, AssertionType assertion,
+        AuditMessageType auditMsg) {
+        auditMsg = getPatientParticipantObjectIdentificationForResponse(response, auditMsg);
+        try {
+            auditMsg = getQueryParamsParticipantObjectIdentificationForResponse(response, auditMsg);
+        } catch (JAXBException ex) {
+            LOG.error("Error while creating ParticipantObjectIdentificationQueryByParameters segment : "
+                + ex.getLocalizedMessage(), ex);
+        }
+        return auditMsg;
+    }
+
+    private AuditMessageType getPatientParticipantObjectIdentificationForRequest(PRPAIN201305UV02 request, AuditMessageType auditMsg) {
+        // Set the Partipation Object Id (patient id)
+        II oII = getPatientIdFromRequest(request);
+        if (oII != null && oII.getRoot() != null && oII.getExtension() != null && !oII.getRoot().isEmpty()
+            && !oII.getExtension().isEmpty()) {
+            createPatientParticipantObjectIdentification(auditMsg, oII.getRoot(), oII.getExtension());
+        }
```

Comment:
The response method for this has an else and with (auditMsg, null, null), should we do that here or should it be removed from the response method?